creme/creme_core/utils/settings.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+
+################################################################################
+#
+# Copyright (c) 2020 Hybird
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+################################################################################
+
+from typing import Union
+
+from django.db import transaction
+
+from creme.creme_core.core.setting_key import SettingKey
+from creme.creme_core.models.setting_value import SettingValue
+
+
+def get_setting_value(key: Union[SettingKey, str], default=None):
+    if not key:
+        return
+
+    try:
+        return SettingValue.objects.get_4_key(key, default=default).value
+    except KeyError:
+        return default
+
+
+def has_setting_value(key: Union[SettingKey, str]):
+    if not key:
+        return False
+
+    key_id = key if isinstance(key, str) else key.id
+    return SettingValue.objects.filter(key_id=key_id).exists()
+
+
+def set_setting_value(key: Union[SettingKey, str], value):
+    if not key:
+        raise KeyError('Empty setting key')
+
+    key_id = key if isinstance(key, str) else key.id
+
+    with transaction.atomic():
+        if value is None:
+            SettingValue.objects.filter(key_id=key_id).delete()
+        else:
+            setting, created = SettingValue.objects.get_or_create(
+                key_id=key_id,
+                defaults={'value': value}
+            )
+
+            if not created:
+                setting.value = value
+                setting.save()
+
+        SettingValue.objects.clear_cache_of(key_id)
```

Comment:
I've made this function for two reasons :
 - Easier tests with TemporarySettingValue
 - Simplify the populate code
As before, I only refactored the concerned populates

Reply:
> As before, I only refactored the concerned populates
See my remark above about "just concerned" bad practice.

As I explained the current set_setting_value() has currently no meaning. Empty the cache is useless in regular code (because it cannot happen to write & read if the same request), & in tests the cache is already emptied between test cases (it's more realistic to write a test case per setting value -- because IRL value is set by a view of creme_config, other parts of the code only read the value).
Why not a 'create_4_key()/create_for_key()' is the manager which only wraps 'get_or_create()'.